pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -19,14 +18,14 @@ import (
 // AnnotationStateHistorian is an implementation of state.Historian that uses Grafana Annotations as the backing datastore.
 type AnnotationStateHistorian struct {
 	annotations annotations.Repository
-	dashboards  dashboards.DashboardService
+	dashboards  *dashboardResolver
```

Comment:
Can't we use the same interface? Then we can use the cache in other places too (i.e images in notifications) without having to change all the places its used?

Reply:
That interface is huge - there'd be 15 additional methods here that are just passthroughs.

Annotations + images never even call the rest of those methods anyway. They only ever look up dashboards by UID. I'd prefer to first refactor both to use a smaller interface, we can call it something like `DashboardReader`.